esp/esp/program/modules/forms/teacherevents.py

```diff
@@ -25,7 +26,7 @@ def save_timeslot(self, program, slot, type):
 
         slot.program = program
         slot.short_description = slot.start.strftime('%A, %B %d %Y %I:%M %p') + " to " + slot.end.strftime('%I:%M %p')
-        slot.description = slot.short_description
+        slot.description = self.cleaned_data['description']
```

Comment:
Looks like `description` is required in the model, so it would be nice to make it optional in the form (i.e. set `blank=True` above), and autofill it if not provided here.